core/src/main/scala/kafka/log/Log.scala

```diff
@@ -272,6 +277,10 @@ class Log(@volatile var dir: File,
     } else {
       if (!dir.getAbsolutePath.endsWith(Log.DeleteDirSuffix)) {
         recoverLog()
+        segmentWithoutTimeIndex.foreach { seg =>
+          if (seg != activeSegment)
```

Comment:
Oh, one more thing. Should we add a comment here saying that this logic can be removed after 0.9 has been deprecated? Not a big deal though.